sn_networking/src/record_store.rs

```diff
@@ -425,21 +423,22 @@ impl NodeRecordStore {
     #[allow(clippy::mutable_key_type)]
     pub(crate) fn store_cost(&self) -> NanoTokens {
         let stored_records = self.records.len();
-        let cost = calculate_cost_for_records(
-            stored_records,
-            self.received_payment_count,
-            self.config.max_records,
-        );
-
-        // vdash metric (if modified please notify at https://github.com/happybeing/vdash/issues):
-        info!("Cost is now {cost:?} for {stored_records:?} stored of {MAX_RECORDS_COUNT:?} max, {:?} times got paid.",
-            self.received_payment_count);
-        NanoTokens::from(cost)
-    }
+        let record_keys_as_hashset: HashSet<Key> = self.records.keys().cloned().collect();
 
-    /// Notify the node received a payment.
-    pub(crate) fn payment_received(&mut self) {
-        self.received_payment_count = self.received_payment_count.saturating_add(1);
+        if let Some(distance_range) = self.distance_range {
```

Comment:
I think the distance_range is not set until we have reached the max num of records inside `prune_storage_if_needed_for_record`. So think we might have to tweak how we set the `distance_range` value.